tests/rest/client/test_relations.py

```diff
@@ -241,6 +241,60 @@ def test_repeated_paginate_relations(self):
         found_event_ids.reverse()
         self.assertEquals(found_event_ids, expected_event_ids)
 
+    def test_pagination_from_sync_messages(self):
+        """Pagination tokens from sync and messages should work."""
+        channel = self._send_relation(RelationTypes.ANNOTATION, "m.reaction", "A")
+        self.assertEquals(200, channel.code, channel.json_body)
+        annotation_id = channel.json_body["event_id"]
+        # Send an event after the relation events.
+        self.helper.send(self.room, body="Latest event", tok=self.user_token)
+
+        # Request sync, but limit the timeline to get a pagination token before
+        # the reaction.
+        filter = urllib.parse.quote_plus(
+            '{"room": {"timeline": {"limit": 1}}}'.encode()
+        )
```

Comment:
Is this 'syncing from the top' as it were, so a later sync would include the 'A' reaction?

Reply:
I'm not sure what you mean by "from the top", this returns the latest events, as a normal sync does.

Reply:
That is confusing to me, since it says the pagination token will be before the reaction; what am I missing?